src/Microsoft.AspNetCore.Http.Abstractions/Resources.resx

```diff
@@ -144,4 +144,13 @@
   <data name="Exception_PortMustBeGreaterThanZero" xml:space="preserve">
     <value>The value must be greater than zero.</value>
   </data>
+  <data name="Exception_UseMiddlewareNoMiddlewareFactory" xml:space="preserve">
+    <value>No service for type '{0}' has been registered.</value>
+  </data>
+  <data name="Exception_UseMiddlewareUnableToCreateMiddleware" xml:space="preserve">
+    <value>Unable to create middleware '{0}'.</value>
```

Comment:
This could be a little more clear about it being the IMiddlewareFactory that's not providing an IMiddleware implementation of type {0}.